core/src/main/scala/dagr/core/execsystem/TaskManager.scala

```diff
@@ -475,6 +478,8 @@ class TaskManager(taskManagerResources: SystemResources = TaskManagerDefaults.de
           logger.exception(e, s"Failed getTasks for [${node.task.name}]: ")
       }
     }
+    // update predecessors just in case some tasks added were already completed
+    updatePredecessors()
```

Comment:
2nd bug (2b): if a completed task is returned by `getTasks`, the dependency graph may not be updated correctly